webapp/components/new-test-results-history-grid.js

```diff
@@ -0,0 +1,220 @@
+const pageStyle = getComputedStyle(document.body);
+
+const PASS_COLOR = pageStyle.getPropertyValue('--paper-green-300');
+const FAIL_COLOR = pageStyle.getPropertyValue('--paper-red-300');
+const NEUTRAL_COLOR = pageStyle.getPropertyValue('--paper-grey-300');
+const COLOR_MAPPING = {
+  // Passing statuses
+  OK: PASS_COLOR,
+  PASS: PASS_COLOR,
+
+  // Failing statuses
+  CRASHED: FAIL_COLOR,
+  ERROR: FAIL_COLOR,
+  FAIL: FAIL_COLOR,
+  NOTRUN: FAIL_COLOR,
+  PRECONDITION_FAILED: FAIL_COLOR,
+  TIMEOUT: FAIL_COLOR,
+
+  // Neutral statuses
+  MISSING: NEUTRAL_COLOR,
+  SKIPPED: NEUTRAL_COLOR,
+  default: NEUTRAL_COLOR,
+};
+
+class TestResultsGrid extends PolymerElement {
+  static get template() {
+    return html`
+        <style>
+          .chart {
+            height: 15rem;
+          }
+          .chart rect, .chart text {
+            cursor: pointer;
+          }
+          .browser {
+            height: 2rem;
+            margin-bottom: -0.5rem;
+          }
+        </style>
+        <h2>
+          <img class="browser" alt="chrome chrome,dev,experimental,master,taskcluster,user:chromium-wpt-export-bot,prod logo" src="/static/chrome-dev_64x64.png">
+          Chrome
+        </h2>
+        <div class="chart" id="chrome"></div>
+
+        <h2>
+          <img class="browser" alt="edge azure,dev,edge,edgechromium,experimental,master,prod logo" src="/static/edge-dev_64x64.png">
+          Edge
+        </h2>
+        <div class="chart" id="edge"></div>
+
+        <h2>
+          <img class="browser" alt="firefox experimental,firefox,master,nightly,taskcluster,user:chromium-wpt-export-bot,prod logo" src="/static/firefox-nightly_64x64.png">
+          Firefox
+        </h2>
+        <div class="chart" id="firefox"></div>
+
+        <h2>
+          <img class="browser" alt="safari azure,experimental,master,preview,safari,prod logo" src="/static/safari-preview_64x64.png">
+          Safari
+        </h2>
+        <div class="chart" id="safari"></div>
+        `;
+  }
+
+  static get properties() {
+    return {
+      dataTable: Object,
+      runIDs: Array,
+    };
+  }
+
+  static get is() {
+    return 'new-test-results-history-grid';
+  }
+
+  constructor() {
+    super();
+
+    // Get the test history data and then populate the chart
+    Promise.all([
+      this.getTestHistory(),
+      this.loadCharts()
+    ]).then(() => this.updateAllCharts(this.historicalData));
+
+    // Google Charts is not responsive, even if one sets a percentage-width, so
+    // we add a resize observer to redraw the chart if the size changes.
+    window.addEventListener('resize', () => {
+      this.updateAllCharts(this.historicalData);
+    });
+  }
+
+  // Load Google charts for test history display
+  async loadCharts() {
+    await window.google.charts.load('current', { packages: ['timeline'] });
+  }
+
+  updateAllCharts(historicalData) {
+    const divNames = [
+      'chrome',
+      'edge',
+      'firefox',
+      'safari'
+    ];
```

Comment:
Nit: It would be more readable to know what this does by giving the ID names that are more descriptive names, e.g. `chromeHistoryChart`